certbot/tests/client_test.py

```diff
@@ -67,8 +67,7 @@ def test_email_retry(self, _rep, mock_get_email):
         mx_err = messages.Error(detail=msg, typ="urn:acme:error:invalidEmail")
         with mock.patch("certbot.client.acme_client.Client") as mock_client:
             mock_client().register.side_effect = [mx_err, mock.MagicMock()]
-            self._call()
-            self.assertEqual(mock_get_email.call_count, 1)
+            self.assertRaises(errors.MissingCommandlineFlag, self._call)
```

Comment:
yeah - hmm open to suggestions on how to preserve that without mock spinning forever (because it'll just keep prompting)